dolphinscheduler-server/src/main/java/org/apache/dolphinscheduler/server/worker/task/dependent/DependentExecute.java

```diff
@@ -104,25 +105,29 @@ private DependResult calculateResultForTasks(DependentItem dependentItem,
                        dependentItem.getDefinitionId(), dateInterval.getStartTime(), dateInterval.getEndTime() );
                 return DependResult.FAILED;
             }
+
             if(dependentItem.getDepTasks().equals(Constants.DEPENDENT_ALL)){
-                result = getDependResultByState(processInstance.getState());
-            }else{
-                TaskInstance taskInstance = null;
-                List<TaskInstance> taskInstanceList = processService.findValidTaskListByProcessId(processInstance.getId());
+                List<TaskNode> taskNodes =
+                        processService.getTaskNodeListByDefinitionId(dependentItem.getDefinitionId());
 
-                for(TaskInstance task : taskInstanceList){
-                    if(task.getName().equals(dependentItem.getDepTasks())){
-                        taskInstance = task;
-                        break;
-                    }
-                }
-                if(taskInstance == null){
-                    // cannot find task in the process instance
-                    // maybe because process instance is running or failed.
-                     result = getDependResultByState(processInstance.getState());
+                if(taskNodes.size() == 0){
```

Comment:
taskNodes. you think must not is empity ??  why  write it  is if( CollectionUtils.isNotEmpty(taskNodes)&&taskNodes.size() == 0){
/////
}